rs/tests/boundary_nodes/rate_limit_canister_test.rs

```diff
@@ -0,0 +1,390 @@
+/* tag::catalog[]
+Title:: Setting rate-limits on the API boundary nodes via rate-limit canister
+
+Goal:: Verify that API boundary nodes dynamically fetch rate-limit configurations from the canister and enforce them for ingress messages.
+
+Runbook:
+. Set up an rate-limit canister.
+. Test that the rate-limit canister API works.
+
+Success:: The rate-limit canister is installed and the API works.
+
+Coverage:: The rate-limit canister interface works as expected.
+
+end::catalog[] */
+
+use anyhow::{bail, Result};
+use candid::{Decode, Encode, Principal};
+use k256::elliptic_curve::SecretKey;
+use rand::{rngs::OsRng, SeedableRng};
+use rand_chacha::ChaChaRng;
+use regex::Regex;
+use slog::{info, Logger};
+use std::env;
+use tokio::runtime::Runtime;
+
+use ic_agent::{
+    identity::{AnonymousIdentity, Secp256k1Identity},
+    Agent, Identity,
+};
+use ic_registry_subnet_type::SubnetType;
+use ic_system_test_driver::{
+    driver::{
+        group::{SystemTestGroup, SystemTestSubGroup},
+        ic::InternetComputer,
+        test_env::TestEnv,
+        test_env_api::{
+            GetFirstHealthyNodeSnapshot, HasPublicApiUrl, HasTopologySnapshot, IcNodeContainer,
+            READY_WAIT_TIMEOUT, RETRY_BACKOFF,
+        },
+    },
+    retry_with_msg_async, systest,
+    util::agent_observes_canister_module,
+};
+use rate_limits_api::{
+    v1::RateLimitRule, AddConfigResponse, DiscloseRulesArg, DiscloseRulesResponse,
+    GetConfigResponse, GetRuleByIdResponse, IncidentId, InitArg, InputConfig, InputRule,
+    OutputRuleMetadata, RuleId, Version,
+};
+
+pub fn config(env: TestEnv) {
```

Comment:
Could you rename this to the more modern:
```suggestion
pub fn setup(env: TestEnv) {
```